### ISSUE TITLE
Fundamental diagram improvements

### ISSUE THREAD

**Issue by gtilg** [patmalcolm91/cell-transmission-model#4]
It would probably be cool to be able to specify the default FD in the yaml for the scenario, or the network.

**Comment by patmalcolm91** [comment]
Yes, this is definitely planned. The following improvements should be implemented:

- [x] Support definition of FD in YAML (and potentially other input formats)
- [ ] Implement other FD formulations besides the simple triangular FD
- [ ] Make FD dynamically adjustable

**Comment by gtilg** [comment]
Sounds good. Regarding the second point, this is doable for piecewise linear FDs. For general FDs, this is a non-trivial task, as far as I know. In fact, I dont know of existing publications regarding CTM with general FDs.
